### ISSUE TITLE
Docs: Update dev.yml file example for GitHub Actions versions

### ISSUE THREAD

**Issue by mjbear** [aristanetworks/avd-workshops#151]
Hello Arista Engineering people,

:bulb: In the markdown there is an example for `dev.yml` that needs updated. _The rest of this will be placed in an issue over in the main AVD repository._

I still had some time on the temporary ATD lab environment and couldn't resist working through the CI/CD section!
In going through that section I discovered that GitHub has made some changes to supported versions of Node.js so some versions of GitHub Actions need adjusted.

**Please test that you too have issues with the GitHub Actions at the pinned versions the lab uses.**
:arrow_right: I'd be glad to submit a PR after receiving your blessing. Thank you!

```
Node.js 16 actions are deprecated. Please update the following actions to use Node.js 20: actions/checkout@v3, actions/setup-python@v3, actions/cache@v3. For more information see: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/.
```

In `dev.yml` I made the following changes:
```
uses: actions/checkout@v3 >> v4

uses: actions/setup-python@v3 >> v5

uses: dorny/paths-filter@v2 >> v3
(in two locations)
```

In `prod.yml` I made the same changes as dev:
```
uses: actions/checkout@v3 >> v4

uses: actions/setup-python@v3 >> v5

uses: dorny/paths-filter@v2 >> v3
(in two locations)
```

**Additionally I've opened up [an issue](https://github.com/aristanetworks/ci-workshops-avd/issues/87) and eventually a PR against the [aristanetworks/ci-workshops-avd](https://github.com/aristanetworks/ci-workshops-avd) repository.**

**Comment by JulioPDX** [comment]
I've been meaning to do this for a while. Great find, please go ahead and submit a PR. Please make sure to update the docs repository (this one) and the [ci fundamentals](https://github.com/aristanetworks/ci-workshops-avd) repo. Great to see you went through the CICD sections, not many take the challenge :)

**Comment by mjbear** [comment]
I've updated the [ci fundamentals](https://github.com/aristanetworks/ci-workshops-avd) via this [PR 89 within that repo](https://github.com/aristanetworks/ci-workshops-avd/pull/89)

I'm working on a PR for the lab docs (this repo).

**Comment by mjbear** [comment]
Submitted PR #152

**Comment by JulioPDX** [comment]
Fixed in #152
